### PR TITLE
Remove lyft api dependency and migrate to lyft->flyteorg 

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Remove lyft api dependency and migrate to lyft->flyteorg.

With upgrading the k8s libraries, cluster resource controller had to change significantly. The client-go interface only accepts client Objects now when using server side apply. This PR borrows heavily from the example [here](https://ymmt2005.hatenablog.com/entry/2020/04/14/An_example_of_using_dynamic_client_of_k8s.io/client-go#Background-Server-Side-Apply) to implement a dynamic server side apply that doesn't require us to unmarshal the templatized yaml files to a known type to comply with the interface changes.

Tested locally on flyte sandbox

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/377

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — APPROVED** [review]
🚢

**Review by EngHabu — APPROVED** [review]
